### PR TITLE
Add Ciena packet-ethernet DOM support

### PR DESCRIPTION
**Pull request by adamkuj**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

---
This change is [Reviewable](https://reviewable.io/reviews/librenms/librenms/7347)

### CODE REVIEW

File: includes/discovery/sensors.inc.php [hunk 1]
```diff
@@ -49,6 +49,10 @@
     include 'includes/discovery/sensors/openbsd.inc.php';
 }
 
+if ($device['os'] == 'ciena-sds') {
```
Comment (laf) [inline includes/discovery/sensors.inc.php]: You don't need this check, please remove this `if()` block

File: includes/discovery/sensors/dbm/ciena.inc.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @author     Adam Kujawski <adamkuj@amplex.net>
+ */
+
```
Comment (laf) [inline includes/discovery/sensors/dbm/ciena.inc.php]: Please remove these additional blank lines, 1 should suffice.

File: includes/discovery/sensors/dbm/ciena.inc.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+
+echo 'Ciena ';
+
+$multiplier = 1;
+$divisor    = 10000;
+foreach ($pre_cache['ciena_oids'] as $index => $entry) {
+    if (is_numeric($entry['wwpLeosPortXcvrRxDbmPower']) && $entry['wwpLeosPortXcvrRxDbmPower'] != 0 && $entry['wwpLeosPortXcvrTxDbmPower'] != 0) {
+        $oid = '.1.3.6.1.4.1.6141.2.60.4.1.1.1.1.105.'.$index;
```
Comment (laf) [inline includes/discovery/sensors/dbm/ciena.inc.php]: Indentations should be four spaces per 'tab'.

File: includes/discovery/sensors/dbm/ciena.inc.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        $descr = dbFetchCell('SELECT `ifDescr` FROM `ports` WHERE `ifName`= ? AND `device_id` = ?', array($index, $device['device_id'])) . ' Rx Power';
```
Comment (laf) [inline includes/discovery/sensors/dbm/ciena.inc.php]: I'm never a fan of this as it relies on port discovery to have run first. Can we not get the port description from within the mib?

File: includes/discovery/sensors/dbm/ciena.inc.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        $limit_low = $entry['wwpLeosPortXcvrLowRxDbmPwAlarmThreshold']/$divisor;
+        $limit = $entry['wwpLeosPortXcvrHighRxDbmPwAlarmThreshold']/$divisor;
+        $current = $entry['wwpLeosPortXcvrRxDbmPower'];
```
Comment (laf) [inline includes/discovery/sensors/dbm/ciena.inc.php]: `$current` should be divided by `$divisor` as well.

File: includes/discovery/sensors/dbm/ciena.inc.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        $entPhysicalIndex_measured = 'ports';
+        discover_sensor($valid['sensor'], 'dbm', $device, $oid, 'rx-'.$index, 'ciena-sds', $descr, $divisor, $multiplier, $limit_low, null, null, $limit, $current, 'snmp', $entPhysicalIndex, $entPhysicalIndex_measured);
+    }
+
+    if (is_numeric($entry['wwpLeosPortXcvrTxDbmPower']) && $entry['wwpLeosPortXcvrTxDbmPower'] && $entry['wwpLeosPortXcvrRxDbmPower']) {
+        $oid = '.1.3.6.1.4.1.6141.2.60.4.1.1.1.1.106.'.$index;
+        $descr = dbFetchCell('SELECT `ifDescr` FROM `ports` WHERE `ifName`= ? AND `device_id` = ?', array($index, $device['device_id'])) . ' Tx Power';
+        $limit_low = $entry['wwpLeosPortXcvrLowTxDbmPwAlarmThreshold']/$divisor;
+        $limit = $entry['wwpLeosPortXcvrHighTxDbmPwAlarmThreshold']/$divisor;
+        $current = $entry['wwpLeosPortXcvrTxDbmPower'];
```
Comment (laf) [inline includes/discovery/sensors/dbm/ciena.inc.php]: Same as before.

File: includes/discovery/sensors/pre-cache/ciena-sds.inc.php [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<?php
+/*
+ * LibreNMS
+ *
+ * Copyright (c) 2016 Neil Lathwood <neil@lathwood.co.uk>
```
Comment (laf) [inline includes/discovery/sensors/pre-cache/ciena-sds.inc.php]: Please use the copyright block from our docs: https://docs.librenms.org/General/Contributing/#copyright

**Comment by laf** [comment]
@adamkuj any comments on the requested changes?

**Comment by laf** [comment]
@adamkuj any update?

**Comment by adamkuj** [comment]
Hi,

I haven't had a chance to make the requested changes yet. Got pulled into
other projects. I had simply copied the junos DOM code and made edits from
there, so any indentation or logic issues were inherited from the existing
code I was aping. I'll try toi take a closer look at some point in the
future.

**Comment by laf** [comment]
@adamkuj Any update on this?

**Comment by laf** [comment]
Please feel free to reply here or push more commits and we can re-open.

**Pull request closed without merging**